### PR TITLE
🎉 New Source:  Omnisend [low-code cdk]

### PR DESCRIPTION
**Pull request by dainiussa**
## What
The following PR adds the new low code CDK-based Omnisend Connector. For more context, please refer to [this issue](https://github.com/airbytehq/connector-contest/issues/180).

## How
This PR uses the lowcode CDK to create a Omnisend connector with the following streams:

- contacts
- campaigns
- carts
- orders
- products

## Recommended reading order
1. `spec.yaml`
2. `omnisend.yaml`
3. schemas

## 🚨 User Impact 🚨
No changes to existing code.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [x] Documentation updated
    - [x] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

## Tests

<details><summary><strong>Acceptance</strong></summary>

acceptance

</details>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
Thanks for the contribution! Someone from the team will take a look as soon as possible :D

File: airbyte-integrations/connectors/source-omnisend/source_omnisend/omnisend.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      $ref: "*ref(definitions.selector)"
+    paginator:
+      type: DefaultPaginator
+      url_base: "*ref(definitions.requester.url_base)"
+      page_size_option:
+        inject_into: "request_parameter"
+        field_name: "limit"
+      pagination_strategy:
+        type: CursorPagination
+        cursor_value: "{{response['paging']['next']}}"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-omnisend/source_omnisend/omnisend.yaml:25]: Do you not need the stop condition?

File: airbyte-integrations/connectors/source-omnisend/source_omnisend/omnisend.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "campaigns"
+      primary_key: "campaignID"
+      path: "/campaigns"
+    retriever:
+      $ref: "*ref(definitions.retriever)"
+      record_selector:
+        extractor:
+          field_pointer: ["campaign"]
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-omnisend/source_omnisend/omnisend.yaml:52]: Did you double check this? It is really `campaign`?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
A few comments but the connector implementation looks solid!

File: airbyte-integrations/connectors/source-omnisend/source_omnisend/omnisend.yaml (hunk 2 above)
Comment (dainiussa) [reply airbyte-integrations/connectors/source-omnisend/source_omnisend/omnisend.yaml:52]: Yes, I did double-check, it is `campaign` 100%

File: airbyte-integrations/connectors/source-omnisend/source_omnisend/spec.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+connectionSpecification:
+  $schema: http://json-schema.org/draft-07/schema#
+  title: Omnisend Spec
+  type: object
+  required:
+    - api_key
+  additionalProperties: true
+  properties:
+    api_key:
+      type: string
```
[the remark this answers is not in the document]
Comment (dainiussa) [reply airbyte-integrations/connectors/source-omnisend/source_omnisend/spec.yaml:11]: I'm sorry not sure what title you want me to add? `title: Omnisend Spec` is on line 4 or is not this one?

[the remark this answers is not in the document]
Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-omnisend/source_omnisend/spec.yaml:11]:
```yaml
api_key:
  title: API Key
  ...
```

File: airbyte-integrations/connectors/source-omnisend/source_omnisend/omnisend.yaml (hunk 1 above)
Comment (dainiussa) [reply airbyte-integrations/connectors/source-omnisend/source_omnisend/omnisend.yaml:25]: When there are no next pages, `response['paging']['next']` is `null`, so not sure if it is necessary to explicitly add the condition? I added it nonetheless, will remove if it is not necessary.

**Review by marcosmarxm — APPROVED** [review]

**Pull request merged**